provider/aws/aws_discover.go

```diff
@@ -105,6 +178,43 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
 		config.Endpoint = &endpoint
 	}
 
+	// Split here for ec2 vs ecs decision tree
+	if service == "ecs" {
+		svc := ecs.New(session.New(), &config)
+
+		log.Printf("[INFO] discover-aws: Filter ECS tasks with %s=%s", tagKey, tagValue)
+		clusterArns, err := getEcsClusters(svc, &ecsCluster)
```

Comment:
If they provide an ECS cluster name, we could skip listing clusters. What do you think?

Reply:
yes! it does seem more idiomatic than searching for a substring in the entire list. Ae you suggesting to go straight to DescribeClusters if the ClusterName is provided? and if-so compiling the ClusterARN with metadata from the ecs_cluster string or just switching the parameter for ecs_cluster_arn to be concise (could be OACAR getting a cluster on a peered network)?

Reply:
I think if the `ecs_cluster` is provided we could go straight to [ListTasks](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ListTasks.html#API_ListTasks_RequestSyntax), since ListTasks supports both cluster name and cluster ARN. Does that seem okay to you?

Then users can pass in either the cluster name or ARN for the `ecs_cluster` parameter.